src/features/surveys/components/SurveySuborgsCard.tsx

```diff
@@ -0,0 +1,48 @@
+import { ChangeEvent } from 'react';
+import messageIds from '../l10n/messageIds';
+import SurveyDataModel from '../models/SurveyDataModel';
+import { useMessages } from 'core/i18n';
+import useModel from 'core/useModel';
+import ZUICard from 'zui/ZUICard';
+
+import { Box, Switch } from '@mui/material';
+
+const SurveySuborgsCard = ({
+  orgId,
+  surveyId,
+}: {
+  orgId: number;
+  surveyId: number;
+}) => {
+  const messages = useMessages(messageIds);
+  const model = useModel((env) => new SurveyDataModel(env, orgId, surveyId));
+  const { data } = model.getData();
+  const orgAccess = data?.org_access === 'sameorg' ? false : true;
+
+  const handleChange = (ev: ChangeEvent<HTMLInputElement>) => {
+    if (ev.target.checked) {
+      model.updateSurveyAccess('suborgs');
+    } else {
+      model.updateSurveyAccess('sameorg');
+    }
+  };
```

Comment:
Why do you prefer having a separate function like this over just writing it inline on the `onChange` property? I prefer the latter for two reasons:

1. It puts the code for _what happens_ close to the code for _when it happens_.
2. It means I don't have to manually specify the type of the event (`ChangeEvent<HTMLInputElement>`) because it can be inferred from the type of the `onChange` property.

Reply:
Totally right! I'll change it!